component/cli/com_redcore/checkpayments.php

```diff
@@ -0,0 +1,149 @@
+ */
+class CheckpaymentsApplicationCli extends JApplicationCli
+{
+	/**
+	 * Status codes
+	 * @var array
+	 */
+	public $finalStatuses = array();
+
+	/**
+	 * Entry point for the script
+	 *
+	 * @return  void
+	 *
+	 * @since   2.5
+	 */
+	public function doExecute()
+	{
+		JFactory::getApplication('site');
+		JPluginHelper::importPlugin('redcore');
+		JPluginHelper::importPlugin('redpayment');
+
+		// Set up statuses that are in their final stage
+		$this->finalStatuses = array(
+			RApiPaymentStatus::getStatusCompleted(),
+			RApiPaymentStatus::getStatusCanceled_Reversal(),
+			RApiPaymentStatus::getStatusDenied(),
+			RApiPaymentStatus::getStatusExpired(),
+			RApiPaymentStatus::getStatusRefunded(),
+			RApiPaymentStatus::getStatusReversed(),
+		);
+
+		$this->out('============================');
+		$this->out('Check Payments status change');
+		$this->out('============================');
+
+		$payments = $this->getPaymentsForChecking();
+
+		$this->out('Number of payments for checking:' . count($payments));
+		$this->out('============================');
+
+		if (!empty($payments))
+		{
+			foreach ($payments as $payment)
+			{
+				// Print out payment info
+				$this->out('============================');
+				$this->out(
+					sprintf('Check for order name "%s" for extension "%s" using "%s" payment plugin:',
+						$payment->order_name,
+						$payment->extension_name,
+						$payment->payment_name
+					)
+				);
+				$this->out('============================');
+
+				// Preform check
+				$status = RApiPaymentHelper::checkPayment($payment->id);
+
+				// Print out status result
+				foreach ($status as $statusKey => $message)
+				{
+					if (is_array($message))
+					{
+						foreach ($status as $key => $value)
+						{
+							$this->out($key . ': ' . $value);
+						}
+					}
+					else
+					{
+						$this->out($statusKey . ': ' . $message);
+					}
+				}
+
+				// Subtract retry count or reset it
+				$paymentNew = RApiPaymentHelper::getPaymentById($payment->id);
+
+				if (!in_array($paymentNew->status, $this->finalStatuses))
+				{
+					// We are still not done, we will subtract retry counter for this payment
+					$paymentNew->retry_counter -= 1;
+
+					RApiPaymentHelper::updatePaymentCounter($paymentNew->id, $paymentNew->retry_counter);
+
+					$this->out('Retry checks left: ' . $paymentNew->retry_counter);
+				}
+
+				$this->out('============================');
+			}
+		}
+
+		$this->out('============================');
+		$this->out('Done !');
+	}
+
+	/**
+	 * Get payments pending for checking
+	 *
+	 * @return mixed
+	 */
+	public function getPaymentsForChecking()
+	{
+		$db = JFactory::getDbo();
+		$finalStatuses = array();
+		$retryTime = RBootstrap::getConfig('payment_time_between_payment_check_requests', 24);
+
+		foreach ($this->finalStatuses as $status)
+		{
+			$finalStatuses[] = $db->q($status);
+		}
+
+		$query = $db->getQuery(true)
+			->select('p.*')
+			->from($db->qn('#__redcore_payments', 'p'))
```

Comment:
Not every payment processor will allow batch processing from cli.  I think there should be a config each plugin itself, and overridable in #__redcore_payment_configuration to disable this functionality for certain plugins/configuration, even if it's active for other plugins at the same time.

One more thing: this means we are keeping all payment data stored (paypal, credit card, etc?)